src/script/ceph-backport.sh

```diff
@@ -77,8 +79,54 @@ function bail_out_github_api {
     false
 }
 
+function blindly_set_pr_metadata {
+    local pr_number="$1"
+    local json_blob="$2"
+    curl --silent --data-binary "$json_blob" 'https://api.github.com/repos/ceph/ceph/issues/'$pr_number'?access_token='$github_token >/dev/null 2>&1 || true
+}
+
+function check_milestones {
+    local milestones_to_check="$(echo "$1" | tr '\n' ' ' | xargs)"
+    info "Active milestones: $milestones_to_check"
+    for m in $milestones_to_check ; do
+        info "Examining all PRs targeting base branch \"$m\""
+        vet_prs_for_milestone "$m"
+    done
+    dump_flagged_prs
+}
+
+function check_tracker_status {
+    local -a ok_statuses=("new" "need more info")
+    local ts="$1"
+    local tslc="${ts,,}"
+    local tslc_is_ok=
+    for oks in "${ok_statuses[@]}"; do
+        if [ "$tslc" = "$oks" ] ; then
+            debug "Tracker status $ts is OK for backport to proceed"
+            tslc_is_ok="yes"
+            break
+        fi
+    done
+    if [ "$tslc_is_ok" ] ; then
+        true
+    else
+        if [ "$tslc" = "in progress" ] ; then
```

Comment:
There might be cases where someone took up a tracker issue, marked as in-progress and is stale now, how should these cases be handled?

Reply:
`--force` option is coming in a follow-up PR

Reply:
(Currently I'm handling that by manually editing the offending fields of the Backport tracker issue to stop the script from complaining.)

Reply:
ah cool